app/controllers/users_controller.rb

```diff
@@ -6,4 +8,29 @@ def index
   def show
     @user = User.find(params[:id])
   end
-end
\ No newline at end of file
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if !user.save
+      # Email is not unique, show flash alert
+      flash[:alert] = "Please try again.  All fields must be filled in and the emails adress must be unique."
+      render 'new'
```

Comment:
This looks good adding the flash message for `Emails` that are not unique and ensuring that all fields are filled out prior to hitting the `Register` button.